grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -684,22 +687,22 @@ class ProductAvailabilityService {
         return data
     }
 
-    Map<InventoryItem, Integer> getQuantityOnHandByInventoryItem(Location location) {
-        def quantityMap = [:]
-        if (location) {
-            def results = ProductAvailability.executeQuery("""
-						select ii, sum(pa.quantityOnHand)
-						from ProductAvailability pa, InventoryItem ii
-						where pa.location = :location
-						and pa.inventoryItem = ii
-						group by ii
-						""", [location: location])
-
-            results.each {
-                quantityMap[it[0]] = it[1]
+    Map<InventoryItem, Integer> getQuantityOnHandByInventoryItem(Location location, List<InventoryItem> inventoryItems = []) {
+        if (!location) {
+            throw new RuntimeException("Must specify location in order to get quantity on hand")
```

Comment:
was there a reason you refactored this part to throw an exception? Maybe there is a case where we'd still want to proceed with e.g. refresh PA, and by doing so, you'd break that operation.

Reply:
From what I have checked it seems we always require a location, but to be on the safe side I think I will revert it to returning an empty array instead of throwing an Exception.